### ISSUE TITLE
Lint in Android Studio highlights the wrong text

### ISSUE THREAD

**Issue by DrPaulT** [flutter/flutter#24790]
In Android Studio the linter is showing the wrong text when highlighting some issues.  Here's a screenshot showing code from one of the apps from the Udacity course and you can see that I have not referenced _format but in fact it has greyed and underlined part of a comment three lines previously.

Not sure what the conditions are to reproduce this as it doesn't seem to happen all the time, but it occurs frequently enough that you can't miss it.

I'm afraid I can't find the exact inspection that causes the error.  Turning off any of the unused method preferences in Android Studio made no difference.

In case there was something up with my Georgia 24 font I switched that back to the default, but it made no difference.

Flutter 30.0.1
Dart 181.5656
Android Studio 3.2.1
Build #AI-181.5540.7.32.5056338, built on October 9, 2018
JRE: 1.8.0_152-release-1136-b06 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.1

<img width="1040" alt="lint_error" src="https://user-images.githubusercontent.com/5172456/49099086-7946be00-f268-11e8-9146-e9e386495e75.png">

**Comment by zoechi** [comment]
Unused declarations are built-in and not configurable.

You could try deleting ` ~/.dartServer/.analysis-driver/` 

Please add the output of `flutter doctor -v`.

**Comment by DrPaulT** [comment]
In fact restarting Android Studio fixed it.  But I can now reproduce it every time:

1.  Start Android Studio.
2. Press alt-cmd-L for Java reformat.

I have attached the output of flutter doctor -v and also the file that is being mis-linted.  Let me know if you need anything else.

[doctor.txt](https://github.com/flutter/flutter/files/2624758/doctor.txt)
[converter_route.dart.txt](https://github.com/flutter/flutter/files/2624766/converter_route.dart.txt)

**Comment by zoechi** [comment]
Have you tried deleting the `.analysis-driver` directory?

**Comment by DrPaulT** [comment]
Yep, I reproduced the defect, deleted the .analysis-driver directory and it did not change anything.

I can now hit alt-cmd-L repeatedly and it cycles through the correct highlight, two wrong ones and then no highlight.

I've attached screenshots to show the cycle.
<img width="1280" alt="s1" src="https://user-images.githubusercontent.com/5172456/49156435-4ad2ec80-f315-11e8-8b1d-c85cd1eabe87.png">
<img width="1280" alt="s2" src="https://user-images.githubusercontent.com/5172456/49156437-4ad2ec80-f315-11e8-8d44-e30fd107ce07.png">
<img width="1280" alt="s3" src="https://user-images.githubusercontent.com/5172456/49156438-4b6b8300-f315-11e8-9be0-978e0d08c20e.png">
<img width="1280" alt="s4" src="https://user-images.githubusercontent.com/5172456/49156440-4b6b8300-f315-11e8-9d14-c48d58156307.png">

**Comment by DrPaulT** [comment]
Thanks for the suggestion.  I've been off Flutter and back on Java for the last few weeks, so haven't had a chance to try the latest release, but am getting back into Dart today.
